cpp/src/cylon/arrow/arrow_comparator.cpp

```diff
@@ -372,6 +372,17 @@ bool TableRowIndexEqualTo::operator()(const int64_t &record1, const int64_t &rec
   return true;
 }
 
+int TableRowIndexEqualTo::compare(const int64_t &record1, const int64_t &record2) const{
+  for (auto &&comp : *idx_comparators_ptr) {
+    if (comp->compare(record1, record2)==0) {
```

Comment:
there are two `comp->compare(record1, record2)` calls here. Shall take it out of the if-else block?